exercises/risk-risiko.cpp

```diff
@@ -27,3 +27,63 @@
   M 3 vs 3 => blue win
   O 2 vs 1 => red win
 */
+
+#define EXIT_SUCCESS 0
+#include <iostream>
+#include <stdlib.h>
+#include <time.h>
+
+using namespace std;
+
+void swap(int *A, int i, int j){
+
+  int tmp = A[i];
+  A[i] = A[j];
+  A[j] = tmp;
+
+}
+
+void sort(int *A, int n){
+
+  for(int i = 0; i < n; i++){
+    for(int j = 0; j < n; j++){
+
+      if(A[i] > A[j]) swap(A,i,j);
```

Comment:
Immagino la task non riguardasse implementare un buon algoritmo di ordinamento, ma non posso non notare che bastano due cambiamenti per migliorare le prestazioni.
In questo momento ci sono dei casi in cui $i = j$, per cui compari un elemento con sè stesso, Per di più, ogni volta scorri l'array per intero nel loop interno.

```c++
// Variante sort
void sort(int *A, int n) {
  for (int i = 0; i < n; i++) {
    for (int j = i + 1; j < n; j++) {
      // Se il nuovo elemento A[j] è più grande di A[i]
      // allora scambiamo i due elementi 
      // per mettere A[j] in cima (a sinistra)
      if (A[j] > A[i]) swap(A, i, j);
    }
  }
}
```